src/tests/functional/frame_navigation_tests.ts

```diff
@@ -24,6 +25,25 @@ test("test frame navigation with exterior link", async ({ page }) => {
   await nextEventOnTarget(page, "frame", "turbo:frame-load")
 })
 
+test("test frame navigation emits fetch-error event when offline", async ({ page }, testInfo) => {
+  testInfo.project.name
+  let browser
+  if (testInfo.project.name === "chrome") {
+    browser = await chromium.launch()
+  } else if (testInfo.project.name === "firefox") {
+    browser = await firefox.launch()
+  } else {
+    throw new Error("unsupported browser")
+  }
+  const context = await browser.newContext()
+  page = await context.newPage()
+  await context.setOffline(true)
```

Comment:
Could we get at this `Context` instance through [Page.context()](https://playwright.dev/docs/api/class-page#page-context)?